### ISSUE TITLE
Would it be possible to see who's in the group?

### ISSUE THREAD

**Issue by Zerolemis** [wmurphyrd/guppe#22]
Thank you for this application. Still, I find it somehow upsetting if I don't know whom the group posts are shared to. So would it be possible to see the suscribers of the group?

**Comment by wmurphyrd** [comment]
Look at the group's "followers" to see the list of everyone who is subscribed. If the list on your local instance is incomplete, click through to view the full profile on https://gup.pe or just go directly to you group profile page, e.g. https://gup.pe/u/guppegroups

Okay I'm just now realizing the web interface doesn't have a way to view the followers list. I'll leave this open as a reminder to do that

**Comment by jibe-b** [comment]
If you search the group from your instance, you may see the followers as usual. For example Mastobikes seen from mastodon.social: https://mastodon.social/web/accounts/1059134/followers

**Comment by behinger** [comment]
I can only see the followers from my instance. This feature would be great!

**Comment by MaPeras** [comment]
Yes, please! This would be wonderful. Especially, as many academics start gathering people on lists manually right now.

**Comment by wmurphyrd** [comment]
I have some concerns about changing this for people who are used to current behavior of not having their presence in the group publicly listed anywhere (consider some popular groups are disability support groups). I'd be interested in an implementation that only listed those who consented (e.g. by sending a command in a DM to the group). 

Guppe is now operating as a platform co-op where supporters make all he decisions about what gets built. Please join and you'll be able to vote for this issue https://opencollective.com/guppe-groups

**Comment by wmurphyrd** [comment]
Estimates for some different approaches:


List group members on Web profile pages: 4 hours
Implement [Mastodon's follower synchronization AP extension](https://docs.joinmastodon.org/spec/activitypub/#follower-synchronization-mechanism) so you can see group members within Mastodon: 8 hours
Implement a user consent system such that only those who sent a special DM to the group would be listed: 8 hours (requires at least a foundation of #12, included in this estimate)

**Comment by francoisMolle** [comment]
I am trying to convince academic colleagues to flock to Mastodon and convene around one @group
But not being able to see who is in  the group is really a BIG drawback that makes it very unattractive: who wants to talk with an anonymous crowd??
This option is a must for Mastodon to be more attractive
Hope this can be done!

**Comment by wmurphyrd** [comment]
@francoisMolle it can be done, if you and others join the co-op to fund the work and vote to implement it: https://opencollective.com/guppe-groups

**Comment by francoisMolle** [comment]
Hi Will

You should also have a one-time pay option too, I think

Cheers

F.M

De : Will Murphy ***@***.***> 
Envoyé : lundi 2 janvier 2023 19:56
À : immers-space/guppe ***@***.***>
Cc : FM ***@***.***>; Mention ***@***.***>
Objet : Re: [immers-space/guppe] Would it be possible to see who's in the group? (#22)

 

@francoisMolle <https://github.com/francoisMolle>  it can be done, if you and others join the co-op to fund the work and vote to implement it: https://opencollective.com/guppe-groups

**Comment by wmurphyrd** [comment]
> Hi Will You should also have a one-time pay option too, I think Cheers F.M

@francoisMolle absolutely we can do work for hire on Guppe, as long as the current co-op members approve the feature. E-mail me at willam@immers.space an I'll get you a quote.

**Comment by francoisMolle** [comment]
I meant : allow people to make on donation rather than monthly fees

 

De : Will Murphy ***@***.***> 
Envoyé : lundi 2 janvier 2023 21:08
À : immers-space/guppe ***@***.***>
Cc : FM ***@***.***>; Mention ***@***.***>
Objet : Re: [immers-space/guppe] Would it be possible to see who's in the group? (#22)

 

Hi Will You should also have a one-time pay option too, I think Cheers F.M

@francoisMolle <https://github.com/francoisMolle>  absolutely we can do work for hire on Guppe, as long as the current co-op members approve the feature. E-mail me at ***@***.*** ***@***.***>  an I'll get you a quote.

**Comment by MaPeras** [comment]
I like your way of handling this. Did you see, that Mastodon Devs put "Groups" on their Public Roadmap? Whatever that means: https://joinmastodon.org/de/roadmap

**Comment by wmurphyrd** [comment]
> I meant : allow people to make on donation rather than monthly fees

@francoisMolle ah my misunderstanding. Sorry. So we're organized as a cooperative, and that means we can collect membership dues rather than donations. In exchange for your dues, you don't just get the continued operation of the platform; you also get decision making power for the platform. However, there's no obligation for you to renew your membership. Some others have chosen to pay annual membership dues once and then cancel. They still get their decision making power for that full year.

**Comment by wmurphyrd** [comment]
> I like your way of handling this. Did you see, that Mastodon Devs put "Groups" on their Public Roadmap? Whatever that means: https://joinmastodon.org/de/roadmap

Thanks @MaPeras! They have more than just roadmapped it; there's a draft implementation available. They are taking a pretty different approach that seems to me to more closely mimic Twitter Communities in that you must choose between posting to a group or to your follows; never both at once. I think we'll find there's room and need for both those groups in Guppe Groups in the fediverse
